addons/binding/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/handler/EnergenieSubdevicesHandler.java

```diff
@@ -0,0 +1,618 @@
+                        thingTypeUID);
+                return;
+        }
+
+        if (state != null) {
+            logger.debug("About to update state for channel {} to {}", channelUID, state.toFullString());
+            updateState(channelUID, state);
+        } else {
+            logger.info("Channel with UID {} won't be updated. Server response is incomplete.", channelUID);
+        }
+    }
+
+    private State getEnergyMonitorState(JsonSubdevice subdevice, String channelID) {
+        State state = UnDefType.UNDEF;
+        switch (channelID) {
+            case EnergenieBindingConstants.CHANNEL_REAL_POWER: {
+                state = new DecimalType(subdevice.getRealPower());
+                break;
+            }
+            case EnergenieBindingConstants.CHANNEL_TODAY_CONSUMPTION: {
+                state = new DecimalType(subdevice.getTodayWh());
+                break;
+            }
+            case EnergenieBindingConstants.CHANNEL_VOLTAGE: {
+                state = new DecimalType(subdevice.getVoltage());
+                break;
+            }
+        }
+        return state;
+    }
+
+    private State getOpenSensorState(JsonSubdevice subdevice, String channelID) {
+        State state = UnDefType.UNDEF;
+        if (channelID.equals(EnergenieBindingConstants.CHANNEL_STATE)) {
+            Integer jsonElement = subdevice.getSensorState();
+            if (jsonElement != null) {
+                return jsonElement == 0 ? OpenClosedType.CLOSED : OpenClosedType.OPEN;
+            } else {
+                state = UnDefType.NULL;
+            }
+        }
+        return state;
+    }
+
+    private State getMotionSensorState(JsonSubdevice subdevice, String channelID) {
+        State state = UnDefType.UNDEF;
+        if (channelID.equals(EnergenieBindingConstants.CHANNEL_STATE)) {
+            Integer value = subdevice.getSensorState();
+            if (value != null) {
+                return value == 0 ? OnOffType.OFF : OnOffType.ON;
+            } else {
+                state = UnDefType.NULL;
+            }
+        }
+        return state;
+    }
+
+    protected void updateThingProperties(JsonSubdevice deviceObj) {
+        Map<String, String> properties = editProperties();
+        properties.put(EnergenieBindingConstants.PROPERTY_DEVICE_ID, Integer.toString(deviceObj.getID()));
+        properties.put(EnergenieBindingConstants.PROPERTY_TYPE, deviceObj.getType().toString());
+        properties.put(EnergenieBindingConstants.PROPERTY_GATEWAY_ID, Integer.toString(deviceObj.getParentID()));
+        updateProperties(properties);
+        logger.debug("Thing properties updated");
+    }
+
+    @Override
+    public void thingUpdated(Thing updatedThing) {
+        ThingUID oldBridgeUID = this.thing.getBridgeUID();
+        ThingUID newBridgeUID = updatedThing.getBridgeUID();
+        if (newBridgeUID != null && !newBridgeUID.equals(oldBridgeUID)) {
+            logger.warn(
```

Comment:
This does not make sense - you are not asked to move the device to another bridge, you are only asked to connect to it through a new Bridge - so do it and decide yourself whether you can access the device that way or not (-> ONLINE/OFFLINE).